tableschema_sql/mapper.py

```diff
@@ -30,10 +32,36 @@ def convert_bucket(self, bucket):
         """
         return self.__prefix + bucket
 
-    def convert_descriptor(self, bucket, descriptor, index_fields=[], autoincrement=None):
+    def has_key_value(self, descriptor, field_name, key_name):
```

Comment:
I don't think this is needed. This is an O(n) key lookup in a dict which should be O(1).